// react-day-picker/dist/esm/useCalendar.js
import { getDates } from "./helpers/getDates.js";
import { getDays } from "./helpers/getDays.js";
import { getDisplayMonths } from "./helpers/getDisplayMonths.js";
import { getInitialMonth } from "./helpers/getInitialMonth.js";
import { getMonths } from "./helpers/getMonths.js";
import { getNavMonths } from "./helpers/getNavMonth.js";
import { getNextMonth } from "./helpers/getNextMonth.js";
import { getPreviousMonth } from "./helpers/getPreviousMonth.js";
import { getWeeks } from "./helpers/getWeeks.js";
import { useControlledValue } from "./helpers/useControlledValue.js";
/** @private */
export function useCalendar(props, dateLib) {
    const today = dateLib.startOfDay(props.today ?? new dateLib.Date());
    const [navigationStartMonth, navigationEndMonth] = getNavMonths(props, dateLib);
    const { startOfMonth } = dateLib;
    const initialDisplayMonth = getInitialMonth(props, dateLib);
    // The first month displayed in the calendar
    const [firstMonth, setFirstMonth] = useControlledValue(initialDisplayMonth, props.month ? startOfMonth(props.month) : undefined);
    /** An array of the months displayed in the calendar. */
    const displayMonths = getDisplayMonths(firstMonth, navigationEndMonth, props, dateLib);
    /** The last month displayed in the calendar. */
    const lastMonth = displayMonths[displayMonths.length - 1];
    /** An array of the dates displayed in the calendar. */
    const dates = getDates(displayMonths, props.endMonth, props, dateLib);
    /** An array of the Months displayed in the calendar. */
    const months = getMonths(displayMonths, dates, props, dateLib);
    /** An array of the Weeks displayed in the calendar. */
    const weeks = getWeeks(months);
    /** An array of the Days displayed in the calendar. */
    const days = getDays(months);
    const previousMonth = getPreviousMonth(firstMonth, navigationStartMonth, props, dateLib);
    const nextMonth = getNextMonth(firstMonth, navigationEndMonth, props, dateLib);
    const { disableNavigation, onMonthChange } = props;
    function isDayDisplayed(day) {
        return weeks.some((week) => {
            return week.days.some((d) => {
                return d.isEqualTo(day);
            });
        });
    }
    function goToMonth(date) {
        if (disableNavigation) {
            return;
        }
        let newMonth = startOfMonth(date);
        // if month is before start, use the first month instead
        if (navigationStartMonth && newMonth < startOfMonth(navigationStartMonth)) {
            newMonth = startOfMonth(navigationStartMonth);
        }
        // if month is after endMonth, use the last month instead
        if (navigationEndMonth && newMonth > startOfMonth(navigationEndMonth)) {
            newMonth = startOfMonth(navigationEndMonth);
        }
        setFirstMonth(newMonth);
        onMonthChange?.(newMonth);
    }
    function goToDay(day) {
        if (isDayDisplayed(day)) {
            return;
        }
        goToMonth(day.date);
    }
    function goToNextMonth() {
        return nextMonth ? goToMonth(nextMonth) : undefined;
    }
    function goToPreviousMonth() {
        return previousMonth ? goToMonth(previousMonth) : undefined;
    }
    const calendar = {
        dates,
        months,
        weeks,
        days,
        today,
        navigationStartMonth: navigationStartMonth,
        navigationEndMonth: navigationEndMonth,
        firstMonth: firstMonth,
        lastMonth,
        previousMonth,
        nextMonth,
        setFirstMonth,
        isDayDisplayed,
        goToMonth,
        goToDay,
        goToNextMonth,
        goToPreviousMonth
    };
    return calendar;
}
