// react-day-picker/dist/esm/useFocus.js
import { useState } from "react";
import { calculateFocusTarget } from "./helpers/calculateFocusTarget.js";
import { getNextFocus } from "./helpers/getNextFocus.js";
/** @private */
export function useFocus(props, calendar, modifiers, selection, dateLib) {
    const { getModifiers } = modifiers;
    const { autoFocus } = props;
    const [lastFocused, setLastFocused] = useState();
    const focusTarget = calculateFocusTarget(calendar, getModifiers, selection.isSelected, lastFocused);
    const [focusedDay, setFocused] = useState(autoFocus ? focusTarget : undefined);
    const blur = () => {
        setLastFocused(focusedDay);
        setFocused(undefined);
    };
    const moveFocus = (moveBy, moveDir) => {
        if (!focusedDay)
            return;
        const nextFocus = getNextFocus(moveBy, moveDir, focusedDay, calendar.navigationStartMonth, calendar.navigationEndMonth, props, dateLib);
        if (!nextFocus)
            return;
        calendar.goToDay(nextFocus);
        setFocused(nextFocus);
    };
    const isFocusTarget = (day) => {
        return Boolean(focusTarget?.isEqualTo(day));
    };
    const useFocus = {
        isFocusTarget,
        setFocused,
        focused: focusedDay,
        setLastFocused,
        blur,
        focusDayAfter: () => moveFocus("day", "after"),
        focusDayBefore: () => moveFocus("day", "before"),
        focusWeekAfter: () => moveFocus("week", "after"),
        focusWeekBefore: () => moveFocus("week", "before"),
        focusMonthBefore: () => moveFocus("month", "before"),
        focusMonthAfter: () => moveFocus("month", "after"),
        focusYearBefore: () => moveFocus("year", "before"),
        focusYearAfter: () => moveFocus("year", "after"),
        focusStartOfWeek: () => moveFocus("startOfWeek", "before"),
        focusEndOfWeek: () => moveFocus("endOfWeek", "after")
    };
    return useFocus;
}
