// react-day-picker/dist/cjs/utils/typeguards.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isDayPickerRange = exports.isDayPickerMultiple = exports.isDayPickerSingle = void 0;
exports.isDateInterval = isDateInterval;
exports.isDateRange = isDateRange;
exports.isDateAfterType = isDateAfterType;
exports.isDateBeforeType = isDateBeforeType;
exports.isSingle = isSingle;
exports.isMulti = isMulti;
exports.isRange = isRange;
exports.isDayOfWeekType = isDayOfWeekType;
/**
 * Returns true if `matcher` is of type `DateInterval`.
 *
 * @group Utilities
 */
function isDateInterval(matcher) {
    return Boolean(matcher &&
        typeof matcher === "object" &&
        "before" in matcher &&
        "after" in matcher);
}
/**
 * Returns true if `value` is a `DateRange` type.
 *
 * @group Utilities
 */
function isDateRange(value) {
    return Boolean(value && typeof value === "object" && "from" in value);
}
/**
 * Returns true if `value` is of type `DateAfter`.
 *
 * @group Utilities
 */
function isDateAfterType(value) {
    return Boolean(value && typeof value === "object" && "after" in value);
}
/**
 * Returns true if `value` is of type `DateBefore`.
 *
 * @group Utilities
 */
function isDateBeforeType(value) {
    return Boolean(value && typeof value === "object" && "before" in value);
}
/**
 * Returns true if the props are for a single selection mode.
 *
 * @group Utilities
 */
function isSingle(props) {
    return props.mode === "single";
}
/**
 * @deprecated This function has been renamed Use `isSingle` instead.
 * @protected
 */
exports.isDayPickerSingle = isSingle;
/**
 * Returns true if the props are for a multiple selection mode.
 *
 * @group Utilities
 */
function isMulti(props) {
    return props.mode === "multiple";
}
/**
 * @deprecated This function has been renamed Use `isMulti` instead.
 * @protected
 */
exports.isDayPickerMultiple = isMulti;
/**
 * Returns true if the props are for a range selection mode.
 *
 * @group Utilities
 */
function isRange(props) {
    return props.mode === "range";
}
/**
 * Returns true if `value` is a `DayOfWeek` type.
 *
 * @group Utilities
 */
function isDayOfWeekType(value) {
    return Boolean(value && typeof value === "object" && "dayOfWeek" in value);
}
/**
 * @deprecated This function has been renamed Use `isRange` instead.
 * @protected
 */
exports.isDayPickerRange = isRange;
