// react-day-picker/dist/cjs/useCalendar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useCalendar = useCalendar;
const getDates_js_1 = require("./helpers/getDates.js");
const getDays_js_1 = require("./helpers/getDays.js");
const getDisplayMonths_js_1 = require("./helpers/getDisplayMonths.js");
const getInitialMonth_js_1 = require("./helpers/getInitialMonth.js");
const getMonths_js_1 = require("./helpers/getMonths.js");
const getNavMonth_js_1 = require("./helpers/getNavMonth.js");
const getNextMonth_js_1 = require("./helpers/getNextMonth.js");
const getPreviousMonth_js_1 = require("./helpers/getPreviousMonth.js");
const getWeeks_js_1 = require("./helpers/getWeeks.js");
const useControlledValue_js_1 = require("./helpers/useControlledValue.js");
/** @private */
function useCalendar(props, dateLib) {
    const today = dateLib.startOfDay(props.today ?? new dateLib.Date());
    const [navigationStartMonth, navigationEndMonth] = (0, getNavMonth_js_1.getNavMonths)(props, dateLib);
    const { startOfMonth } = dateLib;
    const initialDisplayMonth = (0, getInitialMonth_js_1.getInitialMonth)(props, dateLib);
    // The first month displayed in the calendar
    const [firstMonth, setFirstMonth] = (0, useControlledValue_js_1.useControlledValue)(initialDisplayMonth, props.month ? startOfMonth(props.month) : undefined);
    /** An array of the months displayed in the calendar. */
    const displayMonths = (0, getDisplayMonths_js_1.getDisplayMonths)(firstMonth, navigationEndMonth, props, dateLib);
    /** The last month displayed in the calendar. */
    const lastMonth = displayMonths[displayMonths.length - 1];
    /** An array of the dates displayed in the calendar. */
    const dates = (0, getDates_js_1.getDates)(displayMonths, props.endMonth, props, dateLib);
    /** An array of the Months displayed in the calendar. */
    const months = (0, getMonths_js_1.getMonths)(displayMonths, dates, props, dateLib);
    /** An array of the Weeks displayed in the calendar. */
    const weeks = (0, getWeeks_js_1.getWeeks)(months);
    /** An array of the Days displayed in the calendar. */
    const days = (0, getDays_js_1.getDays)(months);
    const previousMonth = (0, getPreviousMonth_js_1.getPreviousMonth)(firstMonth, navigationStartMonth, props, dateLib);
    const nextMonth = (0, getNextMonth_js_1.getNextMonth)(firstMonth, navigationEndMonth, props, dateLib);
    const { disableNavigation, onMonthChange } = props;
    function isDayDisplayed(day) {
        return weeks.some((week) => {
            return week.days.some((d) => {
                return d.isEqualTo(day);
            });
        });
    }
    function goToMonth(date) {
        if (disableNavigation) {
            return;
        }
        let newMonth = startOfMonth(date);
        // if month is before start, use the first month instead
        if (navigationStartMonth && newMonth < startOfMonth(navigationStartMonth)) {
            newMonth = startOfMonth(navigationStartMonth);
        }
        // if month is after endMonth, use the last month instead
        if (navigationEndMonth && newMonth > startOfMonth(navigationEndMonth)) {
            newMonth = startOfMonth(navigationEndMonth);
        }
        setFirstMonth(newMonth);
        onMonthChange?.(newMonth);
    }
    function goToDay(day) {
        if (isDayDisplayed(day)) {
            return;
        }
        goToMonth(day.date);
    }
    function goToNextMonth() {
        return nextMonth ? goToMonth(nextMonth) : undefined;
    }
    function goToPreviousMonth() {
        return previousMonth ? goToMonth(previousMonth) : undefined;
    }
    const calendar = {
        dates,
        months,
        weeks,
        days,
        today,
        navigationStartMonth: navigationStartMonth,
        navigationEndMonth: navigationEndMonth,
        firstMonth: firstMonth,
        lastMonth,
        previousMonth,
        nextMonth,
        setFirstMonth,
        isDayDisplayed,
        goToMonth,
        goToDay,
        goToNextMonth,
        goToPreviousMonth
    };
    return calendar;
}
