// react-day-picker/dist/esm/utils/typeguards.js
/**
 * Returns true if `matcher` is of type `DateInterval`.
 *
 * @group Utilities
 */
export function isDateInterval(matcher) {
    return Boolean(matcher &&
        typeof matcher === "object" &&
        "before" in matcher &&
        "after" in matcher);
}
/**
 * Returns true if `value` is a `DateRange` type.
 *
 * @group Utilities
 */
export function isDateRange(value) {
    return Boolean(value && typeof value === "object" && "from" in value);
}
/**
 * Returns true if `value` is of type `DateAfter`.
 *
 * @group Utilities
 */
export function isDateAfterType(value) {
    return Boolean(value && typeof value === "object" && "after" in value);
}
/**
 * Returns true if `value` is of type `DateBefore`.
 *
 * @group Utilities
 */
export function isDateBeforeType(value) {
    return Boolean(value && typeof value === "object" && "before" in value);
}
/**
 * Returns true if the props are for a single selection mode.
 *
 * @group Utilities
 */
export function isSingle(props) {
    return props.mode === "single";
}
/**
 * @deprecated This function has been renamed Use `isSingle` instead.
 * @protected
 */
export const isDayPickerSingle = isSingle;
/**
 * Returns true if the props are for a multiple selection mode.
 *
 * @group Utilities
 */
export function isMulti(props) {
    return props.mode === "multiple";
}
/**
 * @deprecated This function has been renamed Use `isMulti` instead.
 * @protected
 */
export const isDayPickerMultiple = isMulti;
/**
 * Returns true if the props are for a range selection mode.
 *
 * @group Utilities
 */
export function isRange(props) {
    return props.mode === "range";
}
/**
 * Returns true if `value` is a `DayOfWeek` type.
 *
 * @group Utilities
 */
export function isDayOfWeekType(value) {
    return Boolean(value && typeof value === "object" && "dayOfWeek" in value);
}
/**
 * @deprecated This function has been renamed Use `isRange` instead.
 * @protected
 */
export const isDayPickerRange = isRange;
